Update `WTF::HashTable` to be compatible with `ranges`

c8ef opened the pull request:
#### 5074769715d3de2ee0b47571d82107970abb2f78
<pre>
Update `WTF::HashTable` to be compatible with `ranges`
<a href="https://bugs.webkit.org/show_bug.cgi?id=286548">https://bugs.webkit.org/show_bug.cgi?id=286548</a>

Reviewed by Darin Adler.

I tried replacing `WTF::anyOf` with `std::ranges::any_of`, but encountered
numerous compiler errors, many of which were related to `WTF::HashTable`. The
issue was that, in order to be compatible with ranges, the iterator needed to
satisfy `std::weakly_incrementable`. This patch resolves the issue and includes
related tests.

* Source/WTF/wtf/HashTable.h:
(WTF::HashTableConstIteratorAdapter::operator++):
(WTF::HashTableIteratorAdapter::operator++):
* Tools/TestWebKitAPI/Tests/WTF/HashSet.cpp:
(TestWebKitAPI::TEST(WTF_HashSet, RangesAllAnyNoneOf)):

Canonical link: <a href="https://commits.webkit.org/289446@main">https://commits.webkit.org/289446@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5b126e5eeb8cf0fae048a6dd0029c4ccbc870e88

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/86995 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6505 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41347 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/91853 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/37732 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/89044 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/6781 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/14572 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67243 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25009 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/89997 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5188 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/78746 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47565 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4964 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33121 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/36850 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/79779 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75456 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/33996 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/93740 "Built successfully") | 
| [❌ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/85767 "Found 1 jsc stress test failure: stress/recursive-try-catch.js.dfg-eager-no-cjit-validate") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14156 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10297 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76047 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14359 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74599 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75245 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18512 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19581 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18007 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/7070 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14175 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/19468 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/108261 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/13919 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/26056 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17362 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/15701 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/HashTable.h:
```diff
@@ -1558,7 +1558,7 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(HashTable);
 
         HashTableConstIteratorAdapter& operator++() { ++m_impl; return *this; }
-        // postfix ++ intentionally omitted
+        HashTableConstIteratorAdapter& operator++(int) { auto tmp = *this; ++m_impl; return tmp; }
```
```suggestion
        HashTableConstIteratorAdapter& operator++(int) { auto result = *this; ++m_impl; return result; }
```

c8ef replied:
Done.

darinadler reviewed (approved):
We do prefer words, and "result" is a word while "tmp" is an abbreviation and not a word. But it's fine as is.

c8ef committed:
Update `WTF::HashTable` to be compatible with `ranges`
https://bugs.webkit.org/show_bug.cgi?id=286548

Reviewed by Darin Adler.

I tried replacing `WTF::anyOf` with `std::ranges::any_of`, but encountered
numerous compiler errors, many of which were related to `WTF::HashTable`. The
issue was that, in order to be compatible with ranges, the iterator needed to
satisfy `std::weakly_incrementable`. This patch resolves the issue and includes
related tests.

* Source/WTF/wtf/HashTable.h:
(WTF::HashTableConstIteratorAdapter::operator++):
(WTF::HashTableIteratorAdapter::operator++):
* Tools/TestWebKitAPI/Tests/WTF/HashSet.cpp:
(TestWebKitAPI::TEST(WTF_HashSet, RangesAllAnyNoneOf)):

Canonical link: https://commits.webkit.org/289446@main
